viewing_party/party.py

```diff
@@ -1,23 +1,139 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title == None or genre == None or rating == None:
+        return None
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    index = 0
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            move_movie = movie
+            break
+        index += 1
+    if index > len(user_data["watchlist"]):

```

Comment:
I think there's an off-by-one logical error here. I'm thinking about the scenario where we have a single item in our `watchlist` and it is not the movie we want:

`index` starts as 0, we enter the loop, we do not enter the if-statement, `index` increases to 1, and then we loop around, see there are no more items in `watchlist`, and exit the for-loop.  
We hit line 27 and `index` is 1 which is not greater than the length of `watchlist` which is also 1. Even though we didn't find the movie in the list, we're still going to enter the try block and try to perform the swap.